### PR TITLE
[dist] Don't load rubocop in production

### LINKED ISSUE #5227: %post of obs-api fails (at least on leap 15)
As highlighted in https://github.com/openSUSE/open-build-service/pull/5226, the leap 15 package does not build due to:

```
[ 4229s] obs-api-2.10~alpha.20180626T205514.7228d0879-lp150.5552.1.noarch
[ 4230s] / /
[ 4233s] rails aborted!
[ 4233s] LoadError: cannot load such file -- rubocop/rake_task
[ 4233s] /srv/www/obs/api/lib/tasks/dev.rake:82:in `block (3 levels) in <top (required)>'
[ 4233s] /srv/www/obs/api/lib/tasks/dev.rake:81:in `block (2 levels) in <top (required)>'
[ 4233s] /srv/www/obs/api/lib/tasks/dev.rake:80:in `block in <top (required)>'
[ 4233s] /srv/www/obs/api/lib/tasks/dev.rake:6:in `<top (required)>'
[ 4233s] /srv/www/obs/api/Rakefile:7:in `<top (required)>'
[ 4233s] bin/rails:4:in `require'
[ 4233s] bin/rails:4:in `<main>'
[ 4233s] (See full trace by running task with --trace)
[ 4233s] warning: %post(obs-api-2.10~alpha.20180626T205514.7228d0879-lp150.5552.1.noarch) scriptlet failed, exit status 1

### PR DESCRIPTION
**Pull request by coolo**
Fixes #5227

### CODE REVIEW

**Comment by coolo** [comment]
https://build.opensuse.org/package/show/home:coolo:branches:OBS:Server:Unstable/obs-server

File: src/api/lib/tasks/dev.rake [hunk 1]
```diff
@@ -108,7 +108,7 @@ namespace :dev do
           end
         end
       end
-    end
+    end unless Rails.env.production?
```
Comment (bgeuken) [inline src/api/lib/tasks/dev.rake]: Thanks for the patch:smile: But please move this to the beginning of the :rubocop definition, like
```ruby
unless Rails.env.production?
  namespace :rubocop do
...
end
```

That way it's clear right from the start when this task is available and when it's not.

Comment (coolo) [reply src/api/lib/tasks/dev.rake]: I won't today

Comment (bgeuken) [reply src/api/lib/tasks/dev.rake]: Okay. I solved it

File: src/api/lib/tasks/dev.rake [hunk 2]
```diff
@@ -78,40 +78,56 @@ namespace :dev do
     Rake::Task['db:structure:verify_no_bigint'].invoke
   end
   namespace :lint do
-    namespace :rubocop do
-      require 'rubocop/rake_task'
+    unless Rails.env.production?
```
Comment (hennevogel) [inline src/api/lib/tasks/dev.rake]: :thought_balloon: phew, Gemfile groups baked into code. I don't know guys...

Comment (hennevogel) [reply src/api/lib/tasks/dev.rake]: sure

**Comment by mmohring** [comment]
Fixes also #5222

File: src/api/lib/tasks/dev.rake (hunk 2 above)
Comment (Ana06) [reply src/api/lib/tasks/dev.rake]: @hennevogel what was the problem with this? we have at the moment rake task defined which uses gems that are not installed in production. So, in my opinion we shouldn't define the tasks as well... :thinking:

**Pull request merged**